src/runtime/hvx_sysmon.cpp

```diff
@@ -0,0 +1,44 @@
+#include "runtime_internal.h"
+#include "HalideRuntimeQurt.h"
+
+extern "C" {
+__attribute__((weak)) void HP_profile(unsigned int marker, unsigned char enable);
+WEAK uint32_t halide_qurt_sysmon_enabled = 0;
+WEAK volatile uint32_t halide_qurt_sysmon_marker = 0;
+WEAK volatile uint32_t halide_sysmon_lock = 0;
+
+// Generate a sysmon marker with the specifid ID
+WEAK void halide_sysmon_marker(uint32_t next_marker) {
+    if (!halide_qurt_sysmon_enabled) {
+        return;
+    }
+    if (halide_qurt_sysmon_marker == next_marker) {
+        return;                         // Don't redundantly set marker
+    }
+
+    while (!__sync_bool_compare_and_swap(&halide_sysmon_lock, 0, 1)) { }
+    uint32_t last_marker = halide_qurt_sysmon_marker;   // Get value after lock
+    if (last_marker != next_marker) {   // Only change if we have a new ID
+        halide_qurt_sysmon_marker = next_marker;
+        if (last_marker != 0) {
+            HP_profile(last_marker, 0); // Stop last marker
+        }
+        if (next_marker != 0) {
+            HP_profile(next_marker, 1); // Start new marker
+        }
+    }
+    __sync_synchronize();
+    halide_sysmon_lock = 0;
+}
+
+WEAK void halide_sysmon_start(void) {
```

Comment:
I don't think you need halide_sysmon_start or halide_sysmon_stop, just replace halide_qurt_sysmon_enabled with `HP_profile != NULL`, and remove calls to halide_sysmon_start/stop.